src/hash.jl

```diff
@@ -0,0 +1,80 @@
+# The Base implementation of hash(::Real) is based on Base.decompose
+# to guarantee that all numbers that compare equal are given the same
+# hash. We implement Base.decompose for Mag and Arf.
+function Base.decompose(x::Union{mag_struct,Ptr{mag_struct}})::Tuple{UInt,BigInt,Int}
```

Comment:
since `Base.decompose` is not documented could you write 2-3 lines describing what have you learned?